0.6.0-rc1: Nomad client failure after Docker timeout

### Nomad version
Nomad v0.6.0-rc1 (5187fcfcae50e3dee61e5340eb99805a564b55c3+CHANGES)

### Operating system and Environment details

CentOS 7.3.1611
Docker 17.03.2-ce
Consul v0.8.5

### Issue
Nomad client fails after a number of timed out calls to Docker.

### Nomad Client logs (if appropriate)
```
2017/07/13 22:25:37.902462 [ERR] driver.docker: error removing container: Delete http://unix.sock/containers/915f3abe9950f62ff16c0cd8c376950fff1f8e50038bbd31c867e4783b511d79?force=1&v=1: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
2017/07/13 22:25:37.902624 [INFO] client: task "adpodref-casreader" for alloc "176e696f-22fa-c11b-bef8-2630c61bac1c" completed successfully
2017/07/13 22:25:37.902641 [INFO] client: Not restarting task: adpodref-casreader for alloc: 176e696f-22fa-c11b-bef8-2630c61bac1c
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbbc0fa]
goroutine 301730 [running]:
github.com/hashicorp/nomad/client/driver.(*DockerDriver).Start(0xc424e52240, 0xc4284c2350, 0xc4247b7960, 0xc428161fe0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/nomad/client/driver/docker.go:600 +0x130a
github.com/hashicorp/nomad/client.(*TaskRunner).startTask(0xc42627f200, 0xc4250600c0, 0x0)
	/opt/gopath/src/github.com/hashicorp/nomad/client/task_runner.go:1344 +0x56c
github.com/hashicorp/nomad/client.(*TaskRunner).run(0xc42627f200)
	/opt/gopath/src/github.com/hashicorp/nomad/client/task_runner.go:1031 +0x334
github.com/hashicorp/nomad/client.(*TaskRunner).Run(0xc42627f200)
	/opt/gopath/src/github.com/hashicorp/nomad/client/task_runner.go:561 +0x507
created by github.com/hashicorp/nomad/client.(*AllocRunner).Run
	/opt/gopath/src/github.com/hashicorp/nomad/client/alloc_runner.go:746 +0xbab
```

--- a/client/driver/docker.go
+++ b/client/driver/docker.go
@@ -593,15 +593,18 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (*StartRespon
 			pluginClient.Kill()
 			return nil, fmt.Errorf("Failed to start container %s: %s", container.ID, err)
 		}
+
 		// InspectContainer to get all of the container metadata as
 		// much of the metadata (eg networking) isn't populated until
 		// the container is started
-		if container, err = client.InspectContainer(container.ID); err != nil {
+		runningContainer, err := client.InspectContainer(container.ID)
+		if err != nil {
 			err = fmt.Errorf("failed to inspect started container %s: %s", container.ID, err)
 			d.logger.Printf("[ERR] driver.docker: %v", err)
 			pluginClient.Kill()
 			return nil, structs.NewRecoverableError(err, true)
 		}
+		container = runningContainer
 		d.logger.Printf("[INFO] driver.docker: started container %s", container.ID)
 	} else {
 		d.logger.Printf("[DEBUG] driver.docker: re-attaching to container %s with status %q",

